roles/spawn_container/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+---
+
+- name: Check if mount points exist
+  include_tasks: volume_check.yml
+  with_items:
+    - "{{ volumes }}"
+  when: volumes is defined and (volumes|length>0)
+
+- name: Prepare container log directory
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: root
+    group: root
+    mode: '0750'
+  with_items:
+    - /var/log/containers
+    - "/var/log/containers/{{ name }}"
+
+- name: Container clenup
+  containers.podman.podman_container:
+    name: "{{ name }}"
+    state: absent
+
+- name: Start the container with defined volume mounts
+  containers.podman.podman_container:
+    name: "{{ name }}"
+    image: "{{ image }}"
+    user: "{{ user }}"
+    state: started
+    restart_policy: "always"
+    network: host
+    command: "{{ command }}"
+    volume: "{{ volumes }}"
+    generate_systemd:
+      container_prefix: "osp-observability"
+      path: /etc/systemd/system/
+      restart_policy: always
+  register: result_with_volumes
+  when: volumes is defined and (volumes|length>0)
+
+- name: Start the container without defined volume mounts
+  containers.podman.podman_container:
+    name: "{{ name }}"
+    image: "{{ image }}"
+    user: "{{ user }}"
+    state: started
+    restart_policy: "always"
+    network: host
+    command: "{{ command }}"
+    generate_systemd:
+      container_prefix: "osp-observability"
+      path: /etc/systemd/system/
+      restart_policy: always
+  register: result_no_volumes
+  when: volumes is not defined or (volumes is defined and (volumes|length<0))
+
+- name: Get result
+  set_fact:
+    result: "{% if result_with_volumes.container is defined %}{{ result_with_volumes }}{% else %}{{ result_no_volumes }}{% endif %}"
```

Comment:
Pretty sure you could just register "result"  directly (in both cases) and not bother with this logic

Reply:
Well and yet it didn't work, when I used "result" for both execution branches :)

Reply:
Interesting! When I get this running I'll try it myself and maybe I'll learn something.